packages/connect-webextension/src/contentScript.ts

```diff
@@ -42,4 +42,13 @@ channel.init().then(() => {
             channel.postMessage(event.data, { usePromise: false });
         }
     });
+
+    window.addEventListener('beforeunload', () => {
+        window.postMessage(
+            {
+                type: 'popup-closed',
```

Comment:
not using POPPUP.CLOSE constant is imho _relatively_ correct here, since we want to avoid bundling entire events/popup.ts file and its imports into contentScript.js bundle